src/cuter.erl

```diff
@@ -76,30 +78,55 @@ run(M, F, As, Depth) ->
 
 -spec run(mod(), atom(), input(), depth(), [option()]) -> erroneous_inputs().
 run(M, F, As, Depth, Options) ->
-  Conf = initialize_app(M, F, As, Depth, Options),
-  Mfa = {M, F, length(As)},
-  case pre_run_checks(Mfa) of
-    error -> stop(Conf);
+  Seeds = [{M, F, As, Depth}],
+  run(Seeds, Options).
+
+-spec run([seed()], [option()]) -> erroneous_inputs().
+%% Runs CutEr on multiple units.
+run(Seeds, Options) ->
+  Conf = initialize_app(Options),
+  ConfWithSeeds = add_seeds(Conf, Seeds),
+  Mfas = [{M, F, length(As)} || {M, F, As, _} <- Seeds],
+  case pre_run_checks(Mfas) of
+    error ->
+      stop(ConfWithSeeds);
     ok ->
-      case preprocess(Conf, Mfa) of 
-        false -> stop(Conf);
-        true  -> start(Conf, number_of_pollers(Options), number_of_solvers(Options))
+      case preprocess(ConfWithSeeds, Mfas) of
+        false -> stop(ConfWithSeeds);
+        true  -> start(ConfWithSeeds)
       end
   end.
 
+-spec run_from_file(file:name(), [option()]) -> erroneous_inputs().
+%% Loads the seeds from a file and runs CutEr on all of them.
+%% The terms in the file needs to be in form:
+%%   {M :: module(), F :: atom(), SeedInput :: [input()], Depth :: pos_integer()}.
+run_from_file(File, Options) ->
+  case file:consult(File) of
+    {ok, Seeds} ->
+      run(Seeds, Options);
+    Error ->
+      io:format("Error loading file ~p: ~p~n", [File, Error]),
+      []
```

Comment:
I think that a better option here would be that the function does not return when an error is detected (e.g. use a `throw` or `exit` here instead of returning `[]`).